packages/host/src/lib/host.ts

```diff
@@ -585,6 +589,7 @@ export class Host implements IComponent {
             }, InstanceMessageCode.INSTANCE_ENDED);
 
             if (csic.provides && csic.provides !== "") {
+                // @TODO not sure why we need that
                 csic.getOutputStream()!.unpipe(this.serviceDiscovery.getData(
```

Comment:
As far as I remember it was introduced following @MichalCz comment that after the Instances termites we should detach its output stream being piped to the topic. I know this is pretty much evident from the code and not a detailed technical explanation but unfortunately I don't remember more. With the new topic design perhaps you will need it more than what we needed it back then, when topic could have been used only once.

Reply:
yeah I think it might be to prevent topic closing if the output stream itself has not been ended by the user but only by sequence end. we wil definitely look into that with reusable topics :wink:

Reply:
could be, good luck :muscle: